verification/src/http_server/handlers/verification/sourcify/api.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{VerificationResponse, VerificationResult};
+use actix_web::{error, error::Error};
+use futures::Future;
+use reqwest::Url;
+use std::sync::Arc;
+
+use super::types::{ApiFilesResponse, ApiRequest, ApiVerificationResponse, Files};
+
+#[async_trait::async_trait]
+pub(super) trait SourcifyApi {
+    async fn verification_request(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error>;
+
+    async fn source_files_request(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiFilesResponse, reqwest::Error>;
+}
+
+pub struct SourcifyApiClient {
+    host: Url,
+    request_timeout: u64,
+    verification_attempts: u64,
+}
+
+impl SourcifyApiClient {
+    pub fn new(host: Url, request_timeout: u64, verification_attempts: u64) -> Self {
+        Self {
+            host,
+            request_timeout,
+            verification_attempts,
+        }
+    }
+}
+
+pub async fn make_retrying_request<F, Fut, Response, Error>(
+    verification_attempts: u64,
+    request: F,
```

Comment:
try to remove `F` and make request of type `Fut`, then you can call this with `make_retrying_request(move { /* code */ });`, see https://docs.rs/tokio/latest/tokio/fn.spawn.html